### PR TITLE
Clean up examples to not use toAsync

### PR DESCRIPTION
**Pull request by vkostyukov**
We can just use IO.delay and avoid dealing with TwitterFuture to IO conversions.

### CODE REVIEW

**Comment by joroKr21** [comment]
I don't think that's safe in general - we have to wait for `close` to return

**Comment by vkostyukov** [comment]
> I don't think that's safe in general - we have to wait for `close` to return

I think you're right. I reworked a few things around to make Bootstrap take care of closing the server. This maybe gives Bootstrap too much power so I'm not super sure about this approach yet. Although, I do like the end-user experience working with it, especially where there is no middleware involved.

Take a look at the examples and let me know what do you think about this approach?

File: core/src/main/scala/io/finch/Bootstrap.scala [hunk 1]
```diff
@@ -81,17 +118,27 @@ class Bootstrap[F[_], ES <: HList, CTS <: HList](
...
+
+    val serviced = Dispatcher[F].flatMap { dispatcher =>
+      Resource.make(F.pure(ToService(compiled, dispatcher))) { service =>
+        F.defer(service.close().toAsync[F])
+      }
+    }
+
+    serviced.flatMap { service =>
+      val filtered = filter.andThen(service)
+      Resource.make(F.pure(server.serve(address, filtered))) { listening =>
```
Comment (tpetillot) [inline core/src/main/scala/io/finch/Bootstrap.scala:132]: It is an elegant solution, but it seems to me that there is no input for server configurations. (e.g: `withX` methods on `com.twitter.finagle.Http.Server`)

Comment (vkostyukov) [reply core/src/main/scala/io/finch/Bootstrap.scala:132]: I think what I envisioned is that a final version of the server is passed to Bootstrap.apply. A few of our examples leverage that already (e.g, the streaming one).

Comment (tpetillot) [reply core/src/main/scala/io/finch/Bootstrap.scala:132]: Oh, I missed it! 👍

**Review by tpetillot — COMMENTED** [review]
Seems super handy, LGTM!

**Comment by joroKr21** [comment]
I think it's good to make it easier to use a `Resource`-based app but we shouldn't make it harder to have a custom `main`.

**Comment by vkostyukov** [comment]
> I think it's good to make it easier to use a `Resource`-based app but we shouldn't make it harder to have a custom `main`.

The custom main as in Todo app? Can you elaborate?

**Comment by tpetillot** [comment]
@vkostyukov I think it's about letting the compilation public

**Comment by joroKr21** [comment]
I didn't mean anything specific - just a thought.

**Comment by tpetillot** [comment]
FYI, we released an internal version of this branch and use it in production without any problem since June. @joroKr21

File: core/src/test/scala/io/finch/EndToEndSpec.scala [hunk 2]
```diff
@@ -32,20 +33,24 @@ class EndToEndSpec extends FinchSpec {
     "text/event-stream"
   )
 
-  private def testService[E](service: Resource[IO, Service[Request, Response]])(assertions: Service[Request, Response] => E): E =
-    dispatcherIO.unsafeRunSync(service.use(s => IO(assertions(s))))
+  private object ServiceTest {
+    def apply[CTS] = new {
```
Comment (joroKr21) [inline core/src/test/scala/io/finch/EndToEndSpec.scala:37]: This should be `AnyVal`

File: examples/src/main/scala/io/finch/todo/App.scala [hunk 3]
```diff
@@ -53,7 +53,7 @@ class App(idRef: Ref[IO, Int], storeRef: Ref[IO, Map[Int, Todo]]) extends Endpoi
     }
   }
 
-  final def toService: Resource[IO, Service[Request, Response]] = Bootstrap
+  final val toService: Resource[IO, Service[Request, Response]] = Bootstrap[IO]
```
Comment (joroKr21) [inline examples/src/main/scala/io/finch/todo/App.scala:56]:
```suggestion
  final val service: Resource[IO, Service[Request, Response]] = Bootstrap[IO]
```

File: examples/src/main/scala/io/finch/todo/Main.scala [hunk 4]
```diff
@@ -27,28 +29,37 @@ import io.finch.internal._
...
   private val port: Flag[Int] = flag("port", 8081, "TCP port for HTTP server")
 
-  private def serve(service: Service[Request, Response]): Resource[IO, ListeningServer] =
-    Resource.make(for {
-      srv <- IO(Http.server.withStatsReceiver(statsReceiver))
-      server <- IO(srv.serve(s":${port()}", service))
-    } yield server) { server =>
-      // TODO: We should abstract this out.
-      IO.defer(server.close().toAsync[IO])
+  def app: IO[App] =
```
Comment (joroKr21) [inline examples/src/main/scala/io/finch/todo/Main.scala:36]:
```suggestion
  val app: IO[App] =
```

File: examples/src/main/scala/io/finch/todo/Main.scala [hunk 5]
```diff
@@ -27,28 +29,37 @@ import io.finch.internal._
...
+    for {
+      id <- Ref[IO].of(0)
+      store <- Ref[IO].of(Map.empty[Int, Todo])
+    } yield new App(id, store)
+
+  def serve(a: App): Resource[IO, ListeningServer] =
```
Comment (joroKr21) [inline examples/src/main/scala/io/finch/todo/Main.scala:42]:
```suggestion
  def serve(app: App): Resource[IO, ListeningServer] =
```

File: examples/src/main/scala/io/finch/todo/Main.scala [hunk 6]
```diff
@@ -27,28 +29,37 @@ import io.finch.internal._
...
+    Bootstrap[IO]
+      .serve[Application.Json](a.getTodos :+: a.postTodo :+: a.deleteTodo :+: a.deleteTodos :+: a.patchTodo)
+      .serve[Text.Html](classpathAsset("/todo/index.html"))
+      .serve[Application.Javascript](classpathAsset("/todo/main.js"))
+      .listen(s":${port()}")
+
+  def run(): IO[Unit] =
```
Comment (joroKr21) [inline examples/src/main/scala/io/finch/todo/Main.scala:49]:
```suggestion
  val run: IO[Unit] =
```

**Review by joroKr21 — CHANGES_REQUESTED** [review]
Some cosmetic changes, but also we need to rename some things which would not be binary compatible later. Sorry for the long wait, I was MIA due to personal circumstances. I will address my own comments a bit later 😅

**Review by joroKr21 — APPROVED** [review]

**Pull request merged**